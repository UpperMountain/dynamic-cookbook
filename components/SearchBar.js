import React from 'react';
import {StyleSheet, View, TextInput} from 'react-native';

class SearchBar extends React.Component{
  constructor(props){
    super();
    this.state = {
      text: ''
    };
  }
  
  render(){
    const styles = StyleSheet.create({
      bar: {
        width: 315,
        height: 56,
        backgroundColor: 'rgba(255, 255, 255, .2)',
        borderRadius: 72,
        paddingLeft: 36,
        justifyContent: "center"
      },
      text: {
        fontFamily: 'raleway',
        fontSize: 18,
        color: 'white',
        // lineHeight: 32,
        // flexDirection: 'row'
      }
    });
    
    return(
      <View style={styles.bar}>
        <TextInput style={styles.text} placeholder={'Search Recipes...'} placeholderTextColor={'white'}
        onChangeText={(text) => this.setState({text})} value={this.state.text}/>
      </View>
    );
  }

}

export default SearchBar;