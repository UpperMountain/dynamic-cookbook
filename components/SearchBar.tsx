import React from "react";
import { StyleSheet, View, TextInput, TouchableOpacity } from "react-native";
import { FontAwesome } from "@expo/vector-icons";

const styles = StyleSheet.create({
  bar: {
    height: 50,
    backgroundColor: "rgba(0, 0, 0, .05)",
    borderRadius: 50,
    paddingLeft: 20,
    display: "flex",
    flexDirection: "row",
    alignItems: "center"
  },
  text: {
    fontSize: 18,
    flex: 1
  },
  button: {
    height: "100%",
    width: 50,
    paddingRight: 10,
    display: "flex",
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center"
  }
});

export interface Props {
  value: string;
  onChange: (value: string) => void;
}

interface State {
  open: boolean;
}

class SearchBar extends React.Component<Props, State> {
  state = {
    open: false
  };

  render() {
    const { open } = this.state;
    const { value, onChange } = this.props;

    return (
      <View style={styles.bar}>
        <TextInput
          style={styles.text}
          placeholder="Search Recipes..."
          placeholderTextColor="black"
          onChangeText={e => onChange(e)}
          onFocus={() => this.setState({ open: true })}
          onBlur={() => this.setState({ open: false })}
          value={value}
        />
        {open ? (
          <TouchableOpacity onPress={() => onChange("")} style={styles.button}>
            <FontAwesome name="close" color="black" size={18} />
          </TouchableOpacity>
        ) : (
          <View style={styles.button}>
            <FontAwesome name="search" color="black" size={18} />
          </View>
        )}
      </View>
    );
  }
}

export default SearchBar;
