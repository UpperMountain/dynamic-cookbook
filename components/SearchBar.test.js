import React from "react";
import SearchBar from "./SearchBar";
import renderer from "react-test-renderer";

it("renders without exploding", () => {
  const cmp = renderer.create(<SearchBar />);
});

it("passes snapshot tests", () => {
  const cmp = renderer.create(<SearchBar />);
  expect(cmp.toJSON()).toMatchSnapshot();
});
