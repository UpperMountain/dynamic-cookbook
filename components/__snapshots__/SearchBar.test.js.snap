// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes snapshot tests 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(255, 255, 255, .2)",
      "borderRadius": 72,
      "display": "flex",
      "flexDirection": "row",
      "height": 56,
      "marginTop": 56,
      "paddingLeft": 36,
      "paddingRight": 36,
      "width": 315,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Search Recipes..."
    placeholderTextColor="white"
    style={
      Object {
        "color": "white",
        "flex": 1,
        "fontFamily": "raleway",
        "fontSize": 18,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Text />
</View>
`;
